### ISSUE TITLE
Windows 10 v1809 - No finished video file

### ISSUE THREAD

**Issue by CorSomething** [mikeycal/the-video-editors-render-script-for-blender#13]
Since being upgraded to Windows 10 v1809 the script no longer renders the finished video file.  It just creates two audio files and numerous video files in the script working folder.

**Comment by mikeycal** [comment]
I just tested with v18.03 (OS Build 17134.320). Windows 10 is not presenting with a newer version update than that, so I don't have a way to test v18.09 yet. Having said that, I am finding that the new RANSOMWARE feature on Windows 10 is causing tons of problems in general. 

- Press "Windows KEY" type Security. 
- Click on "Windows Defender Security Center."
- Click "Virus Threat Protection"
- At the bottom of that screen, Click "Ransomeware protection"

If "Controlled Folder Access" is set to "ON", you will notice that programs will have a hard time creating files in your most used folders. You can add FFMPEG and Blender to the ALLOW list of programs, but I noticed that this feature will actually delete files from your computer after they are created. In other words, the Final_video.mp4 file will generate by my script and then "Magically" be deleted by windows OS. 

For now, I am simply turning off the "Controlled Folders Access" option. When it is off, everything works like normal. 

I believe the that Windows 10 October update is actually being held back for most users because of this File delete issue.

**Comment by CorSomething** [comment]
Thanks for your quick reply.

It appears that with this update they have changed the Windows Defender Security Center slightly and it doesn't allow me to change the Controlled Folder Access at present.  I guess I'll just use the blender render until they roll out the working version of this new update.

Thanks for trying to help me though, love the script, it's proven very useful.

**Comment by Chastilian** [comment]
I appear to be having the same issue. I don't know that it is the Windows update creating problems for sure, but I can't think of anything else that's changed on my end. When I run the render script, it renders the video segments correctly, but it appears to get hung up on stitching them back together resulting in exactly what CorSomething describes.

Here is where it gets stuck:
<a href="https://imgur.com/WIdchrP"><img src="https://i.imgur.com/WIdchrP.jpg" title="source: imgur.com" /></a>

I have double checked my directory settings, made sure FFmpeg was up-to-date, and set up the script from scratch again. I also made sure "Controlled Folder Access" was turned off (it was).

Thanks for all the work you've done on the script so far! It's made all the difference.

**Comment by CorSomething** [comment]
I still have this issue but my workaround once the script has run and finished like in your screenshot is as follows:

Open cmd prompt (Win Key+R)

goto folder where "Script Working Folder is located" then:
(for %i in (*.mp4) do @echo file '%i') > mylist.txt

ffmpeg -f concat -i mylist.txt -c copy output.mp4

 ffmpeg -i output.mp4 -i Full_Audio.wav -c:v copy -c:a aac -strict experimental finished-file-name.mp4

**Comment by Chastilian** [comment]
I figured something like that should be possible. Thanks for sharing how you were getting around the issue! You just saved me so much time.

**Comment by mikeycal** [comment]
Here are the final 2 "FFmpeg" commands that should automatically execute after you see "Blender quit" x (Number of Blender Instances).

Assuming that FFmpeg is installed on Windows in "C:\ffmpeg41\" folder and USERNAME is your Windows username.

**This command will read the Concat_Video_List.txt file and join each of the video segments, that are listed, together. (Command is on 1 line)**

`"C:\ffmpeg41\bin\ffmpeg.exe" -f concat -safe 0 -y -i "C:\Users\USERNAME\Script_Working_Folder\Other_Files\Concat_Video_List.txt" -c copy "C:\Users\USERNAME\Script_Working_Folder\AV_Source\Joined_Video.mp4"`

**This command will then MUX the Audio into the Joined Video file. (Command is on 1 line)**

`"C:\ffmpeg41\bin\ffmpeg.exe" -y -i "C:\Users\USERNAME\Script_Working_Folder\AV_Source\Joined_Video.mp4" -y -i "C:\Users\USERNAME\Script_Working_Folder\AV_Source\Full_Audio.m4a" -c:v copy -c:a copy -map 0:v:0 -map 1:a:0 -movflags faststart "C:\Users\USERNAME\Finished_Video.mp4" -async 1`

If you set **auto_delete_temp_files = False** on line 154 of the script you can go to **Script_Working_Folder\Other_Files** folder and view the **render.bat** file to see the commands that did not run. Copy those commands and run them, 1 after the other, from Command Prompt. 

What I think is happening is that Windows 10 doesn't execute the last 2 lines due to some glitch with security that got corrected with a later windows 10 Update. 

As a note: I also experienced this problem with Windows 10 when running the script after a glitchy windows 10 build was installed. My current Windows Build of 1809 **(OS Build 17763.292)** is back to normal. You can check your windows build version by clicking on the WINDOWS MENU icon and typing in winver.exe

**Comment by mattbatt** [comment]
First and foremost MikeyCal Thank you for the videos and thank you for this script.  Were ever on a podcast because your name sounds familiar? 

OK so here's what happened to me.  Re-installed Windows and Blender recently.  My current video editing project was the first one I've tried to render on this new install.  It looked like it was going to take days to render so I googled and found your script.   I tried it once and got a final video of a cube and the audio from my videos.  I figured out that I needed to delete the object in the 3D space so that it would render my video instead.  Suddenly I cant get a final video and it's leaving behind everything in the temp folder even though I have the box checked to delete when done. I also have seen an error that says 
`##############################################
Script_Working_folder is locked by the Operating System.  So it can't be Deleted automatically. 
yadda yadda
###############################
`
I also tried to avoid any of that anti crypto stuff by moving the folder to my F: drive away from all the user stuff. Still no dice. I also white listed blender and ffmpeg. 
I am on build 17763.316 FWIW and the scrip doesn't work in "PowerShell"

**Comment by mikeycal** [comment]
Ok, I finally figured out what was causing the problem. 
I am in the process of updating the code write now. 

The problem was caused by a change to the **start** command in Windows 10 v1809. Microsoft changed the behavior in a way that causes **start** - which is responsible for parallelizing the render - to NOT wait for the mutli-core render process to finish. 

The original code creates a lock that is erased only when blender is done rendering each video segment. Then ffmpeg takes the segments and joins them together. 

the fix was that I had to add a **/max** argument to the **start** command. 

I'm not sure if this is an official change from Microsoft, or if they accidentally broke it. But, for now, I am adding the fix to the latest version of the script.
